client/sftp_client/directory.py

```diff
@@ -819,6 +860,31 @@ def remove_by_file_name(self, name):
 		with self.ftp_handle() as ftp:
 			ftp.remove(name)
 
+	def ftp_read_file(self, file_path):
```

Comment:
Why do these two methods need to be prefixed with `ftp_` instead of just being `(read|save)_file` so they would be exposing the same interface as the local directory. If that were the case then there wouldn't need to be so much duplicate code in `StatusDisplay._save_editor_file`.